### PR TITLE
i#1312 AVX-512 support: Fix API comment and doxygen links.

### PR DESCRIPTION
**Pull request by hgreving2304**
Amends commit 851a961e2d0bedb9254025ba95 with doxygen links and fixes a documentation bug
for MCXT_NUM_OPMASK_SLOTS.

Issue: #1312

### CODE REVIEW

**Comment by hgreving2304** [comment]
Wasn't mentioned in review but I also moved the release docs comments for dr_zmm_t, dr_opmask_t etc. into the non-compatibility section.

File: core/lib/globals_shared.h [hunk 1]
```diff
@@ -1912,8 +1918,9 @@ typedef union _dr_simd_t {
 #        define PRE_XMM_PADDING \
             24 /**< Bytes of padding before xmm/ymm dr_mcontext_t slots */
 #    endif
-#    define MCXT_NUM_OPMASK_SLOTS                                   \
-        8 /**< Number of 16-64-bit OpMask Kn slots in dr_mcontext_t \
+#    define MCXT_NUM_OPMASK_SLOTS                                    \
+        8 /**< Number of 16-64-bit OpMask Kn slots in dr_mcontext_t, \
```
Comment (derekbruening) [inline core/lib/globals_shared.h]: It seems weird to set this to 8 when the real number is 0 as there's nothing in dr_mcontext_t, but I guess this is being staged and the dr_mcontext_t addition will happen before any release or other point where everything must be consistent.

Comment (hgreving2304) [reply core/lib/globals_shared.h]: I will add this to mcontext very soon.

File: api/docs/release.dox [hunk 2]
```diff
@@ -145,9 +145,6 @@ compatibility changes:
    expression and code relying on this needs to be rewritten.
```
Comment (derekbruening) [inline api/docs/release.dox]: > Amends previous commit

Give the hash since it's not the actual previous commit.

Comment (derekbruening) [inline api/docs/release.dox]: > Amends previous commit with doxygen links and a fix for a comment in the public API.

This seems to omit the most important thing here: it fixes a documentation bug for MCXT_NUM_OPMASK_SLOTS.

Comment (hgreving2304) [reply api/docs/release.dox]: The was the part with "and a fix for a comment in the public API."
I'll re-phrase

Comment (hgreving2304) [reply api/docs/release.dox]: Done.

Comment (derekbruening) [reply api/docs/release.dox]: I thought the comment fix was the non-doxygen becoming doxygen for the new type.

**Pull request merged**